os_net_config/impl_ifcfg.py

```diff
@@ -1984,6 +1987,11 @@ def apply(self, cleanup=False, activate=True, config_rules_dns=True):
                         )
                         self.ifdown(interface_name)
                         self.remove_config(ifcfg_file)
+            for nmconn_file in glob.iglob(nm_cleanup_pattern()):
+                interface_name = Path(nmconn_file).stem
+                cmd = ['nmcli', 'connection', 'delete', interface_name]
```

Comment:
instead of `nmcli` can we use nmstate ?

Reply:
You mean use nmstate api or `nmstatectl`? We do use `nmcli`[1] elsewhere right? 

[1] https://github.com/os-net-config/os-net-config/blob/master/os_net_config/impl_nmstate.py#L372

Reply:
@karthiksundaravel As discussed with Rabi, this clean up is required for the interfaces which are having nm connection created during baremetal provisioning.